sklearn/cross_decomposition/_pls.py

```diff
@@ -40,14 +40,24 @@ def _nipals_twoblocks_inner_loop(X, Y, mode="A", max_iter=500, tol=1e-06,
     ite = 1
     X_pinv = Y_pinv = None
     eps = np.finfo(X.dtype).eps
+
+    if mode == "B":
+        # use condition from scipy <=1.2 in pinv2
```

Comment:
Ideally we'd have a better explanation here, it's a bit cryptic. Also, it may be a good idea to include the scipy PR# here.